### PR TITLE
fix: reverted async,path to older versions

### LINKED ISSUE #878: rollback path and async version in pubspec
### Describe the bug

After at_client 3rd party dependencies were upgraded to latest version,  3rd party  packages path and async version are causing an issue in app "flutter pub get" since flutter_test uses lower versions of these packages

### Steps to reproduce

* Upgrade to latest version of at_client(3.0.50) in pubspec and run flutter pub get in at_client_mobile or any app

### Expected behavior

* flutter pub get should run without any errors

### Screenshots

_No response_

### Smartphones

_No response_

### Were you using an atApplication when the bug was found?

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by purnimavenkatasubbu**
**- What I did**
- Reverted the versions of path and async packages to the older versions as it is causing issues during flutter pub get in apps
- dartfmt changes
**- How I did it**

**- How to verify it**
flutter pub get should work without any errors in apps

**- Description for the changelog**

### CODE REVIEW

File: packages/at_client/pubspec.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ environment:
   sdk: '>=2.12.0 <3.0.0'
 
 dependencies:
-  path: ^1.8.3
+  path: ^1.8.2
```
Comment (murali-shris) [inline packages/at_client/pubspec.yaml:18]: path should be 1.8.2 and not ^1.8.2

File: packages/at_client/pubspec.yaml [hunk 2]
```diff
@@ -27,7 +27,7 @@ dependencies:
   archive: ^3.3.5
   http: ^0.13.5
   internet_connection_checker: ^1.0.0+1	
-  async: ^2.10.0
+  async: ^2.9.0
```
Comment (murali-shris) [inline packages/at_client/pubspec.yaml:30]: should be 2.9.0

File: packages/at_client/pubspec.yaml (hunk 1 above)
Comment (gkc) [reply packages/at_client/pubspec.yaml:18]: I disagree.

File: packages/at_client/pubspec.yaml (hunk 2 above)
Comment (gkc) [reply packages/at_client/pubspec.yaml:30]: I disagree

Comment (purnimavenkatasubbu) [reply packages/at_client/pubspec.yaml:30]: Updated

File: packages/at_client/pubspec.yaml (hunk 1 above)
Comment (purnimavenkatasubbu) [reply packages/at_client/pubspec.yaml:18]: updated

Comment (gkc) [reply packages/at_client/pubspec.yaml:18]: I think I misunderstood the context and mistakenly thought Murali's suggestion was to remove the ^

File: packages/at_client/pubspec.yaml (hunk 2 above)
Comment (gkc) [reply packages/at_client/pubspec.yaml:30]: I think I misunderstood the context and mistakenly thought Murali's suggestion was to remove the ^

**Review by gkc — APPROVED** [review]

**Pull request merged**